tests/test_base_models.py

```diff
@@ -1216,4 +1216,15 @@ def main():
         raise Exception("Failed test")
 
 if __name__ == '__main__':
-    main()
+    import cProfile
+    cProfile.run('main()', "output.dat")
+
+    import pstats
+
+    with open("output_time.txt", 'w') as f:
+        p = pstats.Stats("output.dat", stream=f)
+        p.sort_stats("time").print_stats()
+
+    with open("output_calls.txt", 'w') as f:
+        p = pstats.Stats("output.dat", stream=f)
+        p.sort_stats("calls").print_stats()
```

Comment:
This is a great addition for our computational efficiency issues. But I can't remember if we decided it would be useful to have it included in every test file. Let's chat about this as a group.